src/live_restore/live_restore_fs.c

```diff
@@ -1155,27 +1145,34 @@ __wt_live_restore_metadata_to_fh(
     __wt_readlock(session, &lr_fh->bitmap_lock);
     lr_fh->allocsize = lr_fh_meta->allocsize;
     /*
-     * Only allocate a bitmap for a file that has been newly created in the destination and has a
-     * backing source file.
+     * !!!
+     * While the live restore is in progress the bit count reported by in the live restore meta can
+     * hold three states:
+     *  (0)     : This means file has not yet had a bitmap written to the metadata file and
+     *            therefore no relevant writes have gone to the destination.
+     *  (-1)    : This indicates the file has finished migration and the bitmap is therefore empty.
+     *  (nbits) : The number of bits in the bitmap.
      */
-    if (lr_fh->destination.newly_created) {
+    if (lr_fh_meta->nbits == 0 && lr_fh->source_size > 0) {
         uint64_t nbits = lr_fh->source_size / lr_fh_meta->allocsize;
         WT_ERR(__bit_alloc(session, nbits, &lr_fh->destination.bitmap));
         lr_fh->destination.nbits = nbits;
         __wt_readunlock(session, &lr_fh->bitmap_lock);
         return (0);
     }
-    if (lr_fh_meta->nbits != 0) {
+    if (lr_fh_meta->nbits > 0) {
```

Comment:
One more thing I noticed while testing on top of your branch. This if condition is for `> 0`, but the else asserts we must be `nbits == -1`. What happens when we have `nbits == 0`?

Reply:
5ce7a597641ccdc272f95b1b4055fe5f827bfd51